tests/foreman/maintain/test_advanced.py

```diff
@@ -0,0 +1,434 @@
+
+    :expectedresults: Old foreman tasks should delete.
+    """
+    result = sat_maintain.cli.Advanced.run_foreman_tasks_delete(options={'state': 'old'})
+    assert 'FAIL' not in result.stdout
+
+
+def test_positive_foreman_taks_delete_planning(sat_maintain):
+    """Delete planning foreman-tasks using advanced procedure run
+
+    :id: 447f5b05-b6a2-4f3a-8e89-8281995ca1cf
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run foreman-tasks-delete --state planning
+
+    :expectedresults: foreman tasks in planning state should delete.
+    """
+    result = sat_maintain.cli.Advanced.run_foreman_tasks_delete(options={'state': 'planning'})
+    assert 'FAIL' not in result.stdout
+
+
+def test_positive_foreman_taks_delete_pending(sat_maintain):
+    """Delete pending foreman-tasks using advanced procedure run
+
+    :id: 6bd3af66-9910-48c4-8cbb-69c3ddd18d6c
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run foreman-tasks-delete --state pending
+
+    :expectedresults: foreman tasks in pending state should delete.
+    """
+    result = sat_maintain.cli.Advanced.run_foreman_tasks_delete(options={'state': 'pending'})
+    assert 'FAIL' not in result.stdout
+
+
+def test_positive_foreman_task_resume(sat_maintain):
+    """Resume paused foreman-tasks using advanced procedure run
+
+    :id: e9afe55b-e3a4-425a-8bfd-a8df6674e516
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run foreman-tasks-resume
+
+    :expectedresults: foreman tasks in paused state should resume.
+    """
+    result = sat_maintain.cli.Advanced.run_foreman_tasks_resume()
+    assert 'FAIL' not in result.stdout
+
+
+def test_positive_foreman_tasks_ui_investigate(sat_maintain):
+    """Run foreman-tasks-ui-investigate using advanced procedure run
+
+    :id: 3b4f69c6-c0a1-42e3-a099-8a6e26280d17
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run
+        foreman-tasks-ui-investigate
+
+    :expectedresults: procedure foreman-tasks-ui-investigate should work.
+    """
+    result = sat_maintain.cli.Advanced.run_foreman_tasks_ui_investigate(env_var='echo " " | ')
+    assert result.status == 0
+
+
+def test_positive_sync_plan_disable_enable(setup_sync_plan, sat_maintain):
+    """Run sync-plans-enable and sync-plans-disable
+    using advanced procedure run.
+
+    :id: 865df1e1-1189-437c-8451-22d772ff97d4
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run sync-plans-disable
+        2. Run satellite-maintain advanced procedure run sync-plans-enable
+
+    :expectedresults: procedure sync-plans-enable should work.
+    """
+    enable_sync_ids = setup_sync_plan
+    data_yml_path = '/var/lib/foreman-maintain/data.yml'
+    local_data_yml_path = f'{robottelo_tmp_dir}/data.yml'
+
+    result = sat_maintain.cli.Advanced.run_sync_plans_disable()
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
```

Comment:
You could also access the sync plans of setup part and check if they are disabled. We are blind feathing on FM that is always returns correct FAIL/PASS without checking the actual behavior.

Reply:
Same f-m procedures are used in f-m maintenance-mode as well where we test this scenario in detail

Reply:
Same here.

Reply:
its already present, we are good to resolve this thread